Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h

```diff
@@ -24,7 +25,7 @@ Base class for a geometry container for a specific calorimetry subdetector.
 class CaloSubdetectorGeometry {
 
 public:
-  typedef std::vector<CaloCellGeometry const *> CellSet;
+  typedef std::vector<std::shared_ptr<const CaloCellGeometry> > CellSet;
```

Comment:
@bsunanda - using raw pointers is ok here.

Reply:
This will always be event dependent